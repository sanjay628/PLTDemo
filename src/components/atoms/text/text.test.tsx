import React from 'react';
import {render} from '@testing-library/react-native';
import Text from './text.component';
const createTestProps = (props: Object) => ({
  ...props,
});
describe('Text', () => {
  let props: any;
  beforeEach(() => {
    props = createTestProps({});
  });
  it('renders text with default variant and style', () => {
    const {getByText} = render(
      <Text variant="body1" color="red">
        Hello, World!
      </Text>,
    );
    const textElement = getByText('Hello, World!');
    expect(textElement).toBeDefined();
  });

  it('applies custom style and color', () => {
    const {getByText} = render(
      <Text variant="body1" style={{marginBottom: 10}} color="red">
        Custom Text
      </Text>,
    );
    const textElement = getByText('Custom Text');
    expect(textElement).toBeDefined();
  });
  it('renders correctly', () => {
    const tree = render(<Text {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
