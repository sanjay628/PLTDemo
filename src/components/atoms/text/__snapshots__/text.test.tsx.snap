// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 32,
        "fontWeight": "600",
      },
      undefined,
      Object {
        "color": undefined,
      },
    ]
  }
/>
`;
